deep_field_metadetect/metacal.py

```diff
@@ -182,21 +196,75 @@ def metacal_op_shears(obs, reconv_psf=None, shears=None, step=DEFAULT_STEP):
             g2=g2,
         )
         mcal_res[shear] = _render_psf_and_build_obs(
-            mcal_image, obs, reconv_psf, weight_fac=0.5
+            mcal_image,
+            obs,
+            reconv_psf,
+            weight_fac=0.5,
+            max_min_fft_size=max_min_fft_size,
         )
     return mcal_res
 
 
-def match_psf(obs, reconv_psf):
+def match_psf(
+    obs,
+    reconv_psf,
+    return_k_info=False,
+    force_stepk_field=0.0,
+    force_maxk_field=0.0,
+    force_stepk_psf=0.0,
+    force_maxk_psf=0.0,
+    max_min_fft_size=None,
+):
     """Match the PSF on an ngmix observation to a new PSF."""
     wcs = obs.jacobian.get_galsim_wcs()
-    image = get_galsim_object_from_ngmix_obs(obs, kind="image")
-    psf = get_galsim_object_from_ngmix_obs(obs.psf, kind="image")
+    image = get_galsim_object_from_ngmix_obs(
+        obs,
+        kind="image",
+        _force_stepk=force_stepk_field,
+        _force_maxk=force_maxk_field,
+    )
+
+    psf = get_galsim_object_from_ngmix_obs(
+        obs.psf,
+        kind="image",
+        _force_stepk=force_stepk_psf,
+        _force_maxk=force_maxk_psf,
+    )
+
+    if max_min_fft_size is None:
+        ims = galsim.Convolve(
+            [image, galsim.Deconvolve(psf), reconv_psf],
+        )
+
+    else:
+        ims = galsim.Convolve(
+            [image, galsim.Deconvolve(psf), reconv_psf],
+            gsparams=galsim.GSParams(
+                minimum_fft_size=max_min_fft_size, maximum_fft_size=max_min_fft_size
+            ),
+        )
+        ims = ims.withGSParams(
+            minimum_fft_size=max_min_fft_size,
+            maximum_fft_size=max_min_fft_size,
+        )
```

Comment:
Are both of these calls to set the FFT size required? I would have assumed you only needed to set gsparams directly and you did not need the second call.

In any case, can we refactor to?


```suggestion
    ims = galsim.Convolve(
        [image, galsim.Deconvolve(psf), reconv_psf],
    )

    if max_min_fft_size is not None:
        ims = ims.withGSParams(
            minimum_fft_size=max_min_fft_size,
            maximum_fft_size=max_min_fft_size,
        )
```